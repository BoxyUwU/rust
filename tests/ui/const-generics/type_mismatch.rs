fn foo<const N: usize>() -> [u8; N] {
    bar::<N>()
}

fn bar<const N: u8>() -> [u8; N] {}
//~^ ERROR the constant `N` is not of type `usize`

fn main() {}
