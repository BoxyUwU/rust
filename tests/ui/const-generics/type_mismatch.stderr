error: the constant `N` is not of type `usize`
  --> $DIR/type_mismatch.rs:5:26
   |
LL | fn bar<const N: u8>() -> [u8; N] {}
   |                          ^^^^^^^ expected `usize`, found `u8`

error: aborting due to previous error

